tests/metal/texture.slang

```diff
@@ -0,0 +1,642 @@
+//TEST:SIMPLE(filecheck=METAL): -stage compute -entry computeMain -target metal -DEMIT_SOURCE
+//TEST:SIMPLE(filecheck=METALLIB): -stage compute -entry computeMain -target metallib -DEMIT_SOURCE -DMETALLIB
+//TEST:SIMPLE(filecheck=HLSL): -stage compute -entry computeMain -target hlsl -DEMIT_SOURCE
+//TEST:SIMPLE(filecheck=SPIR): -stage compute -entry computeMain -target spirv -emit-spirv-directly -DEMIT_SOURCE
+
+//TEST(compute):COMPARE_COMPUTE_EX(filecheck-buffer=FUNCTIONAL):-slang -compute -dx12 -profile cs_6_6 -use-dxil -shaderobj -output-using-type
+//TEST(compute, vulkan):COMPARE_COMPUTE_EX(filecheck-buffer=FUNCTIONAL):-vk -emit-spirv-directly -compute -shaderobj -output-using-type -render-feature hardware-device
+
+#if !defined(METALLIB)
+    // Metal doesn't support some features, and we need a new test that
+    // can check if the capability system can error them out properly.
+    #define NEED_TO_TEST_FOR_METAL_CAPABILITY 1
+#endif
+
+#if defined(EMIT_SOURCE)
+    // It appears that Slang-test doesn't initialize the depth cube texture
+    // properly.
+    #define TEST_WHEN_DEPTH_CUBE_WORKS
+#endif
```

Comment:
I will create a new issue for this.
The problem is that when the type is depth-cube texture, our "TEST_INPUT" method is not able to set it up properly.